view/sharedcache/core/SharedCache.cpp

```diff
@@ -2794,30 +2819,32 @@ std::vector<std::pair<std::string, Ref<Symbol>>> SharedCache::LoadAllSymbolsAndW
 {
 	std::unique_lock<std::mutex> initialLoadBlock(viewSpecificMutexes[m_dscView->GetFile()->GetSessionId()].viewOperationsThatInfluenceMetadataMutex);
 
+	bool doSave = false;
 	std::vector<std::pair<std::string, Ref<Symbol>>> symbols;
 	for (const auto& img : m_images)
 	{
 		auto header = HeaderForAddress(img.headerLocation);
-		std::shared_ptr<MMappedFileAccessor> mapping;
-		try {
-			mapping = MMappedFileAccessor::Open(m_dscView, m_dscView->GetFile()->GetSessionId(), header->exportTriePath)->lock();
-		}
-		catch (...)
-		{
-			m_logger->LogWarn("Serious Error: Failed to open export trie %s for %s", header->exportTriePath.c_str(), header->installName.c_str());
-			continue;
-		}
-		auto exportList = SharedCache::ParseExportTrie(mapping, *header);
-		std::vector<std::pair<uint64_t, std::pair<BNSymbolType, std::string>>> exportMapping;
+		auto exportList = GetExportListForHeader(*header, [&]() {
+			try {
+				auto mapping = MMappedFileAccessor::Open(m_dscView, m_dscView->GetFile()->GetSessionId(), header->exportTriePath)->lock();
+				doSave = true;
```

Comment:
Is this the only difference between the various "open a file" lambdas passed to `GetExportListForHeader`? If so you could simplify these call sites by having `GetExportListForHeader` take `bool* didModifyExportList = nullptr`. Callers that are interested in knowing that `m_exportInfos` was modified could pass a pointer to bool in. Others would omit it. This would let you move the file opening logic into `GetExportListForHeader`.

Reply:
I would have done this but the call in `SharedCache::InitializeHeader` has a different way of loading a mapped file.

`bool* didModifyExportList = nullptr` - I didn't do this just cause it seemed this is the only instance where it mattered so it seemed 50/50 whether the extra parameter made sense or what I did. For future stuff though it might make more sense to have the parameter so its clear to a caller that they need to do a save if they didn't know.

Reply:
Ah, I missed that `SharedCache::InitializeHeader` is doing something different. Bummer!